test/dom/get-role.test.ts

```diff
@@ -326,11 +319,165 @@ describe('getRole', () => {
     ['var', { given: ['<var></var>', 'var'], want: NO_CORRESPONDING_ROLE }],
     ['video', { given: ['<video></video>', 'video'], want: NO_CORRESPONDING_ROLE }],
     ['wbr', { given: ['<wbr></wbr>', 'wbr'], want: NO_CORRESPONDING_ROLE }],
+
+    // Custom elements
     ['custom element', { given: ['<x-button>My Button</x-button>', 'x-button'], want: 'generic' }],
     [
       'custom element (with role)',
       { given: ['<x-button role="button">My Button</x-button>', 'x-button'], want: 'button' },
     ],
+
+    // SVG
+    ['svg', { given: ['<svg></svg>', 'svg'], want: 'graphics-document' }],
+    ['svg[role=img]', { given: ['<svg role="img"></svg>', 'svg'], want: 'img' }],
+    [
+      'svg[role=graphics-symbol img]',
+      { given: ['<svg role="graphics-symbol img"></svg>', 'svg'], want: 'graphics-symbol' },
+    ],
+    ['animate', { given: ['<svg><animate></animate></svg>', 'animate'], want: 'none' }],
+    ['animateMotion', { given: ['<svg><animateMotion></animateMotion></svg>', 'animateMotion'], want: 'none' }],
+    [
+      'animateTransform',
+      { given: ['<svg><animateTransform></animateTransform></svg>', 'animateTransform'], want: 'none' },
+    ],
+    ['circle', { given: ['<svg><circle /></svg>', 'circle'], want: 'none' }],
+    ['circle (empty title)', { given: ['<svg><circle><title></title></circle></svg>', 'circle'], want: 'none' }],
```

Comment:
cc @cmorten Added tests that attempt to satisfy the behavior outlined in the [SVG AAM Including elements in the accessibility tree](https://www.w3.org/TR/svg-aam-1.0/#include_elements) section. `circle` got tests for every behavior that describes; other similar elements like `rect`, `ellipse`, `path`, etc. follow the same codepath so it’s assumed they match